iset.mm

```diff
@@ -95229,6 +94988,217 @@ gives an ordered pair whose first element is the index (translated
+        opeq2d frecsuc fveq2i 3eqtr4g eqtrd simpr frec2uzsucd exp31 finds mpcom
+        wss opeq1d a2d ) KUNSAKFTZUDTZKJTZUEZQAUAUFZFTZUDTZYEJTZUEZUGAUHFTZUDTZ
+        UHJTZUEZUGAUBUFZFTZUDTZYNJTZUEZUGAYNUIZFTZUDTZYSJTZUEZUGAYDUGUAUBKYEUHU
+        EZYIYMAUUDYGYKYHYLUUDYFYJUDYEUHFUJUKYEUHJUJULUMYEYNUEZYIYRAUUEYGYPYHYQU
+        UEYFYOUDYEYNFUJUKYEYNJUJULUMYEYSUEZYIUUCAUUFYGUUAYHUUBUUFYFYTUDYEYSFUJU
+        KYEYSJUJULUMYEKUEZYIYDAUUGYGYBYHYCUUGYFYAUDYEKFUJUKYEKJUJULUMAEDUOZUDTZ
+        EYKYLAEUPSZDGSZUUIEUELMEDUPGUQURAYJUUHUDAYJUHBCEUSTZHBUFZUTVAVBZUUMCUFZ
+        IVBZUOZVCZUUHVDZTZUUHUHFUUSPVEAUUJUUKUUTUUHUEZLMUUJUUKVHUUHVFSUVAEDUPGV
+        IUUHUURVFVGVJURVKUKABEJLRVLVMYNUNSZAYRUUCUVBAYRUUCUVBAVHZYRVHZYQUTVAVBZ
+        YQYOVNTZIVBZUOZUDTZUVEUUAUUBUVDUVEUULSZUVGGSZUVIUVEUEUVDYQUULSZUVJUVDUN
+        UULYNJAUNUULJVOZUVBYRAUNUULJVPUVMABEJLRVQUNUULJVSVJVRUVBAYRVTZWAZEYQWIV
+        JZUVDYQUUOIVBZGSZUVKCGUVFUUOUVFUEZUVQUVGGUUOUVFYQIWBZWCUVDUUPGSZCGWDZUV
+        RCGWDBUULYQUUMYQUEZUWAUVRCGUWCUUPUVQGUUMYQUUOIWEZWCWFAUWBBUULWDZUVBYRAU
+        WABCUULGOWGZVRUVOWHUVDYOUULGWRZSZUVFGSUVDUNUWGYNFAUNUWGFVOUVBYRABCDEFGH
+        ILMNOPWJVRUVNWAZYOUULGWKVJZWHZUVEUVGUULGUQURUVDYTUVHUDUVDYTYQUVFUOZUURT
+        ZUVHUVDYTYPUVFUOZUURTZUWMUVDYTYOUURTZUWOUVDYSUUSTZYNUUSTZUURTZYTUWPUVDU
+        CUFZUURTZUWGSZUCUWGWDUUHUWGSZUVBUWQUWSUEUVDUXBUCUWGUVDUWTUWGSZVHZUXAUWT
+        UDTZUWTVNTZUOZUURTZUWGUXEUWTUXHUURUXDUWTUXHUEUVDUWTUULGWLWMUKUXEUXIUXFU
+        TVAVBZUXFUXGIVBZUOZUWGUXEUXIUXFUXGUURVBZUXLUXFUXGUURWNUXEUXFUULSZUXGHSU
+        XLUWGSZUXMUXLUEUXDUXNUVDUWTUULGWOWMZUXEGHUXGAGHXRZUVBYRUXDNWPUXDUXGGSUV
+        DUWTUULGWKWMZWQUXEUXJUULSZUXKGSZUXOUXEUXNUXSUXPEUXFWIVJUXEUXFUUOIVBZGSZ
+        UXTCGUXGUUOUXGUEZUYAUXKGUUOUXGUXFIWBZWCUXEUWBUYBCGWDBUULUXFUUMUXFUEZUWA
+        UYBCGUYEUUPUYAGUUMUXFUUOIWEZWCWFAUWEUVBYRUXDUWFWPUXPWHUXRWHUXJUXKUULGWS
+        URZBCUXFUXGUULHUUQUXLUURUXJUYAUOUWGUYEUUNUXJUUPUYAUUMUXFUTVAWEUYFWTUYCU
+        YAUXKUXJUYDXHUURXAZXBXCXDUYGXEXEXFAUXCUVBYRAEUULSZUUKUXCAUUJUYILEXGVJME
+        DUULGWSURVRUVNUCUUHYNUWGUURXIXCYSFUUSPVEYOUWRUURYNFUUSPVEXJXKUVDYOUWNUU
+        RUVDUWHYOUWNUEUWIYOUULGWLVJUKXLUVDUWNUWLUURUVDYPYQUVFUVCYRXMXSUKXLUVDUW
+        MYQUVFUURVBZUVHYQUVFUURWNUVDUVLUVFHSUVHUWGSZUYJUVHUEUVOUVDGHUVFAUXQUVBY
+        RNVRUWJWQUVDUVJUVKUYKUVPUWKUVEUVGUULGWSURBCYQUVFUULHUUQUVHUURUVEUVQUOUW
+        GUWCUUNUVEUUPUVQUUMYQUTVAWEUWDWTUVSUVQUVGUVEUVTXHUYHXBXCXDXLUKUVDBYNEJA
+        UUJUVBYRLVRRUVNXNVMXOXTXPXQ $.
+    $}
+
+    ${
+      $d C v x y $.  $d C x y $.  $d F x y $.  $d R v $.  $d R x y $.
+      $d G x y $.  $d S x y $.  $d T x y $.  $d ph v x y $.  $d ph x y $.
+      frecuzrdgdom.2 $e |- G = frec ( ( x e. ZZ |-> ( x + 1 ) ) , C ) $.
+      $( The domain of the result of the recursive definition generator on
+         upper integers.  (Contributed by Jim Kingdon, 24-Apr-2022.) $)
+      frecuzrdgdom $p |- ( ph -> dom ran R = ( ZZ>= ` C ) ) $=
```

Comment:
This theorem does not depend on G!? Why is frecuzrdgdom.2 needed?

Reply:
It is used in the proof. Although it would, of course, be possible to use `frec ( ( x e. ZZ |-> ( x + 1 ) ) , C )` throughout the proof, I think this sort of "local definition" usually makes for a more readable situation (especially in a case like this where something like a third or a half of the steps have a `G` in them).

Did you have another idea of how to handle this situation? I could imagine doing things with naming or comments or something, but perhaps there is a possibility I hadn't thought of.

Reply:
It can be left as is, or be a lemma followed by the theorem, proved from that lemma and eqid, as in http://us2.metamath.org/mpeuni/bj-pwcfsdom.html
I would favor the latter, but there are many such "proof-facilitating hypotheses" in set.mm and apparently it is not cared much about leaving these hypotheses.

Reply:
OK, I see. Are there really many such cases in set.mm? Either I did not use such theorems before, or I did not notice that. For myself, I often use local definitions as "proof-facilitating hypotheses", but always provide at least one theorem/lemma which contains this defintion explicitly. Therefore, I would prefer Benoît's suggestion.

Reply:
There are certainly examples in this work of "lemma followed by the theorem" style, such as  http://us.metamath.org/ileuni/freccllem.html / http://us.metamath.org/ileuni/freccl.html or http://us.metamath.org/ileuni/frecsuclem.html / http://us.metamath.org/ileuni/frecsuc.html . Now basically `frecuzrdgdom` and `frecuzrdgfun` are lemmas for `frecuzrdgtclt` , and `frecuzrdgsuct` is a lemma for `iseqp1t` so I don't think this is fundamentally different. If they are lemmas should they have `lem` in the name? Well, maybe, but the names here are already getting long and I'm not sure I have especially brilliant ideas about how to name any of this. Some of the considerations are (1) set.mm has some of the same naming questions (for example, whether `uzrdg` is a good fragment of a name for the "pre-seq" theorems as now used in both set.mm and iset.mm), (2) there are several generations of these theorems, some of which may go away at some time and some of which might be worth keeping side by side (and no, I don't have super firm opinions about which are which), (3) there are more changes coming up mostly likely, at least if I (or someone else) takes this work further, (4) at least some of these "lemmas" are used in more than one (but less than, say, five) theorems which I guess argues for some kind of name not including `lem`.

Reply:
> Therefore, I would prefer Benoît's suggestion.

Give me a day or so and I'll update this pull request to do that.

Reply:
done